src/Datadog.Trace/Tracer.cs

```diff
@@ -95,6 +95,19 @@ internal Tracer(TracerSettings settings, IAgentWriter agentWriter, ISampler samp
                 }
             }
 
+            if (Settings.GlobalSamplingRate != null)
```

Comment:
What's the rationale for a default setting to not use a GlobalSamplingRate of 1? I don't know much about the sampling rate stuff

Reply:
because we want to ensure the default behavior is to use the existing "rate by service" sampling (agent controlled sampling).

So for now, in order for a user to get 100% (sample rate of 1) they have to manually specify this option.

Reply:
Ok. I could use some further brush-up on the sampling decisions we're making now and in the future because it's still not really clear to me, but the code looks absolutely fine